receiver/redisreceiver/metadata.yaml

```diff
@@ -5,8 +5,55 @@ attributes:
     description: Redis CPU usage state
   db:
     description: Redis database identifier
+  version:
+    description: Redis binary version
+  role:
+    description: Redis node's role, replica or primary
+  cmd:
+    description: Redis command name
 
 metrics:
+  redis.maxmemory:
+    enabled: true
+    description: The value of the maxmemory configuration directive
+    unit: By
+    gauge:
+      value_type: int
+
+  redis.version:
+    enabled: true
+    description: Redis binary version, metric value is fixed to be 1
+    gauge:
+      value_type: int
+    attributes: [version]
+
+  redis.role:
+    enabled: true
+    description: Redis node's role, 0 means replica, 1 means primary
+    gauge:
+      value_type: int
+    attributes: [role]
```

Comment:
If you have an attribute named `role` (that I assume is either `primary` or `replica`), do you even need the 1 and 0 as the Gauge value? It should be sufficient to just look at the attribute, and the gauge value can always be 1 (changing the meaning from "if its `1`, its `primary`, regardless of the attribute" to "there is 1 instance of type `primary` running").

Reply:
With the gauge value, consumer doesn't have to translate the attribute if they want to present the role change on a graph.

Reply:
But they would have to know that `0` means `replica` and `1` means `primary`, right?

Reply:
`redis.role{state="primary"}` would easily show you how the node's role change over time.

On the principle, we all agree that an *Enum* type would be awesome. However, there is no timeseries backend that I know of that support an *Enum* type natively. So our discussion becomes pointless as all exporters are going to convert to a simple number, and lose the mapping information (especially if the *enum* has many values, it can't even be specified in the *Unit* field)

That's why I conceded in [my PR](https://github.com/open-telemetry/opentelemetry-specification/pull/2518) that the best way to represent *Enums* was using *Attributes* (also what [OpenMetrics describes as StateSet](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#stateset)).

Reply:
Don't worry, *UpDownCounter* is converted to *Gauge* by most exporters (it's the case with Prometheus exporters).

Reply:
I mean for now I can't define stateset type metric because redisreceiver uses mdatagen to auto generate metrics functions. And mdatagen only supports sum and gauge metric types.

Reply:
> they'd have to know that, but we can document this clearly and when they consume this metric, they will know what 0 and 1 mean.

I don't think we can enforce this. I also think most backends do not allow you to map the value to an Enum (see @bertysentry 's comment above). Also, you can use attributes, set the value to `1` and get precisely what you propose, but in a way in which most metrics backends will be able to deal with it. 

Therefore, I don't see the upside in forcing backends to support this and forcing users to copy the definitions to their backends manually, especially when there is a way of doing it where you can just specify the values on the fly on the metric. 

The stateset in this case would be represented as a Sum, with different states as attributes, if I understand correctly.